### PR TITLE
Document how to get backtrace with gdb at boot from any service

### PR DESCRIPTION
**Pull request by LaakkonenJussi**
Added documentation about how to get backtrace from services that crash at boot. This requires only gdb and the relevant debuginfo packages to be installed and shows straightforward way for service debug with gdb.

### CODE REVIEW

**Review by vigejolla — APPROVED** [review]
LGTM

File: Reference/Sailfish_OS_Cheat_Sheet/README.md [hunk 1]
```diff
@@ -326,6 +326,54 @@ Enabled=0
...
+set height 0
+set verbose off
+set logging file <path_to_log_file>
+set logging on
+handle SIG33 pass nostop noprint
+run
+backtrace full
+```
+
+And add a systemd drop-in file to `/usr/lib/systemd/system/<service_name>.service.d/override.conf` with the content to override the existing service file variables:
```
Comment (Thaodan) [inline Reference/Sailfish_OS_Cheat_Sheet/README.md:352]: If using devel-su -c 'systemctl edit <service>' you can avoid the manual steps to create the directory, the files and system daemon reload.

File: Reference/Sailfish_OS_Cheat_Sheet/README.md [hunk 2]
```diff
@@ -326,6 +326,54 @@ Enabled=0
...
 
+### System service debugging
+
+Sometimes changes to system services that are being started at boot cause unwanted results and the service crashes. Or the service crashes only on device boot. These can be resolved by checking the backtraces of such service. This set of instructions applies generally for other services as well.
+
+This requires only `gdb` from the `tools` repository:
+```ssu ar tools
+pkcon refresh
+pkcon install gdb
+```
```
Comment (Thaodan) [inline Reference/Sailfish_OS_Cheat_Sheet/README.md:337]: You can use `pkcon refresh && pkcon install gdb` for easier refresh.

**Review by Thaodan — COMMENTED** [review]
All in all looks good to me, only some smaller suggestions.
I'm wondering is there a bug for this?

File: Reference/Sailfish_OS_Cheat_Sheet/README.md (hunk 1 above)
Comment (LaakkonenJussi) [reply Reference/Sailfish_OS_Cheat_Sheet/README.md:352]: I need to try this. I was not aware of that, I usually do the things the hard way.

File: Reference/Sailfish_OS_Cheat_Sheet/README.md (hunk 2 above)
Comment (LaakkonenJussi) [reply Reference/Sailfish_OS_Cheat_Sheet/README.md:337]: I'd rather use `zypper ref tools` but I guess `pkcon` is preferred usually as it is installed already on the devices.

File: Reference/Sailfish_OS_Cheat_Sheet/README.md [hunk 3]
```diff
@@ -326,6 +326,54 @@ Enabled=0
 Enabled=0
 ```
 
+### System service debugging
+
+Sometimes changes to system services that are being started at boot cause unwanted results and the service crashes. Or the service crashes only on device boot. These can be resolved by checking the backtraces of such service. This set of instructions applies generally for other services as well.
+
+This requires only `gdb` from the `tools` repository:
+```ssu ar tools
```
Comment (vigejolla) [inline Reference/Sailfish_OS_Cheat_Sheet/README.md:334]: I just now noticed that you had the first line of commands on the same line as backticks. This does not work, they need to be on separate lines. The same error is on all instances where you added code blocks.

And then I noticed that you used the tools repo, which I believe doesn't work on release devices. I think gdb is found on sdk repo there. I don't have any release device at hand so I can't easily verify though.

**Review by vigejolla — CHANGES_REQUESTED** [review]

File: Reference/Sailfish_OS_Cheat_Sheet/README.md (hunk 1 above)
Comment (LaakkonenJussi) [reply Reference/Sailfish_OS_Cheat_Sheet/README.md:352]: Actually, `systemctl edit` creates the overrides to `/etc/systemd/system/` and then the changes made to the location we use by default (`/usr/lib/systemd/system/`) do not work afterwards for the service. Apparently one would have to manually remove the `/etc/systemd/system/<servicename>.service.d/*.conf`. I think manual steps here are more obvious. Usually systemd tools just obfuscate things (I do not trust them).

File: Reference/Sailfish_OS_Cheat_Sheet/README.md (hunk 2 above)
Comment (LaakkonenJussi) [reply Reference/Sailfish_OS_Cheat_Sheet/README.md:336]: Changed to oneliner.

File: Reference/Sailfish_OS_Cheat_Sheet/README.md (hunk 1 above)
Comment (LaakkonenJussi) [reply Reference/Sailfish_OS_Cheat_Sheet/README.md:352]: Also, when and if we change the location where `systemctl edit` makes the changes we'd have then outdated location here in the docs. By telling the exact location here and forcing to do a bit of manual labor the details are always permanent. This isn't really something one needs to do even on monthly, let alone weekly basis.

File: Reference/Sailfish_OS_Cheat_Sheet/README.md (hunk 3 above)
Comment (LaakkonenJussi) [reply Reference/Sailfish_OS_Cheat_Sheet/README.md]: Right, I guess the format details were not correct. Made the changes to format.

And changed to use the sdk repo. Although on devel devices it does not seem to have `gdb`..

Comment (LaakkonenJussi) [reply Reference/Sailfish_OS_Cheat_Sheet/README.md]: Oh, no, apparently `gdb` is in sdk repo. Nevermind that.

**Comment by LaakkonenJussi** [comment]
> LGTM, but please squash the commits

Will do, always :).

File: Reference/Sailfish_OS_Cheat_Sheet/README.md (hunk 1 above)
Comment (Thaodan) [reply Reference/Sailfish_OS_Cheat_Sheet/README.md:352]: > Actually, `systemctl edit` creates the overrides to `/etc/systemd/system/` and then the changes made to the location we use by default (`/usr/lib/systemd/system/`) do not work afterwards for the service. Apparently one would have to manually remove the `/etc/systemd/system/<servicename>.service.d/*.conf`. I think manual steps here are more obvious. Usually systemd tools just obfuscate things (I do not trust them).

the changes can be revert with systemctl revert.

Comment (LaakkonenJussi) [reply Reference/Sailfish_OS_Cheat_Sheet/README.md:352]: We could separately raise this issue in the documentation with the warning that `revert` will cancel all changes and the changes are stored elsewhere than what we commonly use as location when `edit` is used. Perhaps a reference later on then to that section somewhere in this added documentation is in place. Therefore, resolving this.

Comment (Thaodan) [reply Reference/Sailfish_OS_Cheat_Sheet/README.md:352]: > We could separately raise this issue in the documentation with the warning that `revert` will cancel all changes and the changes are stored elsewhere than what we commonly use as location when `edit` is used. Perhaps a reference later on then to that section somewhere in this added documentation is in place. Therefore, resolving this.

I wasn't aware of that, will check.

**Pull request merged**